### PR TITLE
[Merged by Bors] - feat(Data/Fin/Tuple/Basic): `repeat_comp_rev`

### PR DESCRIPTION
**Pull request by technosentience**
Prove `repeat_comp_rev`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/Tuple/Basic.lean
Comment (urkud) [inline Mathlib/Data/Fin/Tuple/Basic.lean:10]: How many filed do these 2 imports add to the closure?

File: Mathlib/Data/Fin/Tuple/Basic.lean
Comment (urkud) [inline Mathlib/Data/Fin/Tuple/Basic.lean:452]: Non-closing `simp`s aren't allowed in Mathlib. Please use `simp?` to turn it into `simp only` (or into` simp? [] says ...`).

Comment (technosentience) [reply Mathlib/Data/Fin/Tuple/Basic.lean]: Fixed.

File: Mathlib/Data/Fin/Tuple/Basic.lean
Comment (technosentience) [reply Mathlib/Data/Fin/Tuple/Basic.lean:10]: Not sure I'm looking at the right thing, but the graph produced by `lake exe graph --to Mathlib.Data.Fin.Tuple.Basic --reduce` went from 560 edges to 644.

Comment (urkud) [reply Mathlib/Data/Fin/Tuple/Basic.lean]: I pushed a proof that doesn't use `zify`. I guess, now we need a review from another maintainer.
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
